.pre-commit-config.yaml

```
@@ -0,0 +1,40 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+
+# exclusions should be separated with a pipe (|) character and a newline
+exclude: |
+  (?x)^(
+      tests/expected/.*
+  )$
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
```

mixed line endings are mostly a non-issue these days, but maybe this can help?

```suggestion
      - id: trailing-whitespace
      - id: end-of-file-fixer
      - id: mixed-line-ending
        args:
          - --fix=lf
```